src/tokenizer.ts

```diff
@@ -560,470 +703,523 @@ export class Tokenizer extends DiagnosticEmitter {
     while (pos < end) {
       this.tokenPos = pos;
       let c = text.charCodeAt(pos);
-      switch (c) {
-        case CharCode.CARRIAGERETURN: {
-          if (!(
-            ++pos < end &&
-            text.charCodeAt(pos) == CharCode.LINEFEED
-          )) break;
-          // otherwise fall-through
+      if (c <= 0x7F) {
```

Comment:
You are losing me down here, as I can't really see from the diff what has changed, just that a lot has changed. For reviewing purposes, I would prefer if these changes can be split into smaller PRs.